### DISCUSSION TITLE
Requesting Formulas

### DISCUSSION THREAD

**Discussion by justinormont** [Homebrew/discussions#4356] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

How/where does one request a formula?

Specifically I'm looking for `zpaqfranz`[^1], which is a fork of `zpaq`[^2].

The original `zpaq` is abandoned for 7 years, and `zpaqfranz` is actively maintained with the last release two weeks ago.

[^1]: https://github.com/fcorbelli/zpaqfranz
[^2]: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/zpaq.rb

**Comment by ZhongRuoyu** [comment]
Please read https://docs.brew.sh/Adding-Software-to-Homebrew. We will review a pull request.

**Reply by justinormont** [reply]
Thanks though that answers the question of "How do I create and submit my own formula?". Instead I'm looking if there's an existing place to simply post a request for a formula. E.g. an idea board of useful tools which are missing, and can be taken up by any interested party.

**Reply by SMillerDev** [reply]
Yes, pull requests. If you are interested you can make one.

**Comment by alebcay** [comment, marked as answer]
Since I had some time, I opened a PR for `zpaqfranz` in https://github.com/Homebrew/homebrew-core/pull/126686 and it has been merged. It can be installed with `brew install zpaqfranz`.

Generally speaking, please still refer to the guidance at https://docs.brew.sh/Adding-Software-to-Homebrew re: opening a pull request. There's no guarantee or assurances at all that any sort of request for a formula here will be fulfilled.

Maybe there's a suitable place to put such requests for the community to contribute desired formuale, but I don't think this is necessary a good place for requests, since it could get spammy.

**Reply by justinormont** [reply]
You're awesome. Thanks so much. 

I agree, this is not the right place for requests. I was hoping a suitable place existed. Perhaps just a separate discussion category, which would be filtered out by non-interested folks. Beyond discussions, I think choices would be a wiki or issues.

**Comment by fcorbelli** [comment]
Can I thanks both of you in the source of the next release?
Like that
```
18 Thanks to https://github.com/justinormont            for the proposed MacOS's homebrew install
19 Thanks to https://github.com/alebcay                 for acceptance of homebrew install
```

**Reply by justinormont** [reply]
Sounds good. @alebcay may want to write something more specific, though I might edit as:
```
18 Thanks to https://github.com/justinormont            for the proposed Homebrew install formula for macOS and x64 Linux
19 Thanks to https://github.com/alebcay                 for coding the Homebrew install formula for macOS and x64 Linux
```

Beyond the macOS build, I also tested under x64 Ubuntu 20. The `zpaqfranz autotest -all` command runs successfully.

**Reply by fcorbelli** [reply]
The "full" test is
```
zpaqfranz autotest -all -to /tmp/somewhere
```

```
root@aserver:~ # zpaqfranz autotest -all -to /tmp/somewhere
zpaqfranz v57.5m-JIT-L, (23 Mar 2023)
franz:-all                                      4
Self-test for correct internal functioning
Creating autotest folder in <</tmp/somewhere/>>
Iteration 0/9 chunksize    1.000.000
(...)
The test batchfile is: /tmp/somewhere/dotest.sh

110.069 seconds (000:01:50) (all OK)
```
This will extract (~2GB) of test files

Then run the .script (/tmp/somewhere/dotest.sh), looking to the output (some interation is needed by pauses)

Basically, it verifies the compliancy with both big-little endian and memory allocation (on sparc64 you need appropriate alignment switches), extraction and compression and hashing  

If the script works, at 99% all is OK, there should be no problems even on 'strange' platforms (as strange as Mac PPC, yep zpaqfranz runs on PPC too , currently I am working on 1993 Sparcstation :)

Of course there should be "OK" everywhere (no WARNING or ERROR)

Inside braces what you should look as "four grand total test" (1), (2), (3) and (4)
```
Renamed     files  0
NOT renamed files  256

0.125 seconds (00:00:00) (all OK)
(1) You should read NOT renamed files  256
```

```
|SHA-256: 67708591460BCE3BC45AE086A342F9F390AD2913A22639EC7AF3646B7D2AEA78 [          1.000.000]     |/tmp/somewhere/verifica

0.013 seconds (00:00:00) (all OK)
(2) You should read SHA-256: 67708591460BCE3BC45AE086A342F9F390AD2913A22639EC7AF3646B7D2AEA78
```

```
SURE    :       42.727 of       42.727 (stored=decompressed=file on disk)
All OK (paranoid test with check against filesystem)

70.814 seconds (000:01:10) (all OK)
(3) You should read SURE    :       42.727 of       42.727 (stored=decompressed=file on disk)
```

```
GLOBAL SHA256: 7B32DB3F6F0180062773C5A046A2C99D17BE7668202379BA136E7A4D134FFC26

3.812 seconds (000:00:03) (all OK)
(4) You should read GLOBAL SHA256: 7B32DB3F6F0180062773C5A046A2C99D17BE7668202379BA136E7A4D134FFC26
```


After the test the **/tmp/somewhere** can be deleted

**Reply by justinormont** [reply]
Ok, left macOS results on the PR -- https://github.com/fcorbelli/zpaqfranz/pull/48#issuecomment-1484910492
